build.js

```diff
@@ -616,461 +891,536 @@ function copyIcons(app, outputPath) {
+    if (shortName === null) {
+      throw new BuildError("Build aborted: No shortName in the manifest");
+    }
 
-  var apkBuildName = "";
-  if (!config.debug) {
-    if (skipSigning) {
-      apkBuildName = shortName + "-release-unsigned.apk";
+    var apkBuildName = "";
+    if (!config.debug) {
+      if (skipSigning) {
+        apkBuildName = "app-release-unsigned.apk";
+      } else {
+        apkBuildName = "app-release-aligned.apk";
+      }
     } else {
-      apkBuildName = shortName + "-aligned.apk";
+      apkBuildName = "app-debug.apk";
     }
-  } else {
-    apkBuildName = shortName + "-debug.apk";
-  }
 
-  if (!app.manifest.android) {
-    logger.warn('you should add an "android" key to your app\'s manifest.json',
-                'for android-specific settings');
-    app.manifest.android = {};
-  }
+    if (!app.manifest.android) {
+      logger.warn('you should add an "android" key to your app\'s manifest.json',
+        'for android-specific settings');
+      app.manifest.android = {};
+    }
 
-  return Promise.try(function createAndroidProjectFiles() {
+    return Promise.try(function createAndroidProjectFiles() {
       if (!config.repack) {
         return createProject(api, app, config);
       }
     })
-    .then(function copyResourcesToProject() {
-      return [
-        copyIcons(app, config.outputPath),
-        copyMusic(app, config.outputPath),
-        copyResDir(app, config.outputPath),
-        copySplash(api, app, config.outputPath),
-        copyAssets(api, app, config.outputPath)
-      ];
-    })
-    .all()
-    .then(function buildAPK() {
-      if (!skipAPK) {
-        if (config.repack) {
-          return repackAPK(api, config.outputPath, apkBuildName);
-        } else {
-          var antScheme = (config.debug ? "debug" : "release");
-          return spawnWithLogger(api, 'ant', [antScheme], {
-              cwd: config.outputPath
-            });
-        }
-      }
-    })
-    .then(function () {
-      if (!skipSigning) {
-        return signAPK(api, shortName, config.outputPath, config.debug);
-      }
-    })
-    .then(function () {
-      if (!skipAPK) {
-        return moveAPK(api, app, config, apkBuildName)
-          .tap(function (apkPath) {
-            logger.log("built", chalk.yellow(config.packageName));
-            logger.log("saved to " + chalk.blue(apkPath));
-          })
-          .then(function (apkPath) {
-            if (argv.reveal) {
-              require('child_process').exec('open --reveal "' + apkPath + '"');
-            }
+      .then(function copyResourcesToProject() {
+        return [
+          // changed from config.outputPath to gradle project path
+          copyIcons(app, projectPath + "/app/src/main"),
+          copyMusic(app, projectPath + "/app/src/main"),
+          copyResDir(app, projectPath + "/app/src/main"),
+          copySplash(api, app, projectPath + "/app/src/main"),
+          copyAssets(api, app, projectPath + "/app/src/main")
```

Comment:
concat of projectPath + "/app/src/main" has been used in multiple places. it can be assign to another variable and use it.

Reply:
Done